<?xml version="1.0"?>
<launch>
    <!--
    FoundationPose ROS Launch File

    Usage:
        roslaunch foundation_pose foundation_pose.launch server_ip:=192.168.1.100

    With RealSense:
        roslaunch foundation_pose foundation_pose.launch server_ip:=192.168.1.100 launch_realsense:=true
    -->

    <!-- Arguments -->
    <arg name="server_ip" default="localhost" doc="GPU server IP address"/>
    <arg name="port" default="5555" doc="ZeroMQ port"/>
    <arg name="rate_limit" default="10.0" doc="Max processing rate (Hz)"/>
    <arg name="jpeg_quality" default="80" doc="JPEG compression quality"/>
    <arg name="publish_tf" default="true" doc="Publish TF transform"/>
    <arg name="frame_id" default="camera_color_optical_frame" doc="Camera frame ID"/>
    <arg name="object_frame_id" default="object" doc="Object frame ID for TF"/>

    <!-- RealSense launch option -->
    <arg name="launch_realsense" default="false" doc="Launch RealSense node"/>

    <!-- RealSense Camera (optional) -->
    <group if="$(arg launch_realsense)">
        <include file="$(find realsense2_camera)/launch/rs_camera.launch">
            <arg name="align_depth" value="true"/>
            <arg name="enable_pointcloud" value="false"/>
            <arg name="enable_sync" value="true"/>
        </include>
    </group>

    <!-- FoundationPose Node -->
    <node name="foundation_pose" pkg="foundation_pose" type="ros_client.py" output="screen">
        <param name="server_ip" value="$(arg server_ip)"/>
        <param name="port" value="$(arg port)"/>
        <param name="rate_limit" value="$(arg rate_limit)"/>
        <param name="jpeg_quality" value="$(arg jpeg_quality)"/>
        <param name="publish_tf" value="$(arg publish_tf)"/>
        <param name="frame_id" value="$(arg frame_id)"/>
        <param name="object_frame_id" value="$(arg object_frame_id)"/>
    </node>

</launch>
